hypothesis-python/docs/tutorial/adapting-strategies.rst

```diff
@@ -0,0 +1,135 @@
+Adapting strategies
+===================
+
+This page discusses ways to adapt strategies to your needs, either by transforming them inline with |strategy.map|, or filtering out unwanted inputs with |strategy.filter| and |assume|.
+
+Mapping strategy inputs
+-----------------------
```

Comment:
I'd love to include the ordered-pair example here, just so we don't only show how to do that with composite.

It might also be nice to show combining map with filter/assume, to "generate, fixup, and discard if we can't fix" (a fairly general and usually efficient pattern).